sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
`URL#setURLStreamHandlerFactory` can only be called once per JVM. If we use `FsUrlStreamHandlerFactory` directly, we won't be able to support other factories. I wrapped it for future extendability.